packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -123,8 +124,19 @@ export const GridColumnHeaderItem = React.memo(
       [publish],
     );
 
+    const classNames = [classes?.columnHeader];
+
+    if (column.headerClassName) {
+      classNames.push(
+        clsx(
+          isFunction(column.headerClassName)
```

Comment:
Shouldn't we drop this `isFunction` helper? What problem does it solve? I fail to understand it, there is already a built-in primitive in JavaScript it does this.

It also seems to

1. creates friction when writing new code, we have to figure out the write import path, base on the location of the file we want to use it into
2. creates an indirection that prevents clearly understand what it sets as a constraint on the public API.

```suggestion
          typeof column.headerClassName === 'function'
```

Reply:
`isFunction` seems to be used in quite a lot of places already. We had a similar discussion I think a couple of months ago regarding potentially unneeded helper functions. We never reached an agreement regarding this. We can always reopen the discussion and have a decision this time since it is inevitable that we will pick it up again at some point.

Reply:
I'm in favour of keeping it, it's an easy helper. And it does more than the `typeof...`. I do not understand the frictions.
![image](https://user-images.githubusercontent.com/936978/119985344-d00cea00-bfc2-11eb-8652-743a1e006bb8.png)

Reply:
> It does more than the `typeof`

@dtassone Your example is demonstrating another value of removing the `isFunction` helper. it's telling you that the block is a dead code and should be removed.

I could find documentation about it in the documentation of TypeScript: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

> having to define a function to figure out if a type is a primitive is kind of **a pain**.